Go/Go.sublime-syntax

```diff
@@ -118,26 +121,72 @@ contexts:
     #
     # //go:some_directive arg0 arg1 ...
     #
+    # These have been adopted outside the gc compiler for use in linting
+    # directives and pragmas for other compiler suites where they take the
+    # more generalised form of:
+    #
+    # //namespace:some_directive ...
+    #
     # They're not part of the language spec, may not be recognized by some Go
     # compilers, and may stop working in future releases. Therefore,
     # highlighting them as compiler pragmas could be misleading. We scope them
     # as plain comments by default, and add some detailed meta scopes for
     # enterprising users wishing to add more color.
-    - match: (//)(go)(:)({{ident}})?
+    - match: (//)([a-z]+[a-z-]*)(:)({{directive}})[ ]?
       captures:
         1: punctuation.definition.comment.go
         2: meta.keyword.annotation.go
         3: meta.punctuation.accessor.colon.go
         4: meta.variable.function.go
       push:
         - meta_scope: meta.annotation.go comment.line.go
+        - match: .+
+          scope: meta.variable.parameter.go
+        # End the annotation scope at EOL, but stretch the comment scope
+        # indefinitely to the right.
+        - match: $
+          set: pop-line-comment
+
+    # Go also has line renumbering; line directives specify the source
+    # position for the character immediately following the comment as
+    # having come from the specified file, line and column. Both single
+    # line and block comment line directives are valid.
+    - match: (//)(line) ([a-zA-Z0-9. :]*(?::[0-9]+){1,2})$
+      captures:
+        1: punctuation.definition.comment.go
+        2: meta.variable.function.go
+        3: meta.variable.parameter.go
+      push:
+        - meta_scope: meta.annotation.go comment.line.go
+        - match: $
+          set: pop-line-comment
+
+    - match: (/\*)(line) ([a-zA-Z0-9. :]*(?::[0-9]+){1,2})(\*/)
+      captures:
+        1: punctuation.definition.comment.go
+        2: meta.variable.function.go
+        3: meta.variable.parameter.go
+        4: punctuation.definition.comment.go
+      push:
+        - meta_scope: meta.annotation.go comment.block.go
         - match: \S+
           scope: meta.variable.parameter.go
         # End the annotation scope at EOL, but stretch the comment scope
         # indefinitely to the right.
         - match: $
           set: pop-line-comment
```

Comment:
Pushing into a context after a closed block quote doesn't seem to make sense.

`/*line foo.bar::9 */ ???annotation here???` 

I don't think arbritary text after a block quote is parsed as part of a comment or annotation, is it?

Reply:
Yes, this didn't feel right at the time, but I was unsure of how to achieve what I wanted.

Reply:
Hopefully this most recent change fixes this. If not, please make a suggestion.